synapse/config/emailconfig.py

```diff
@@ -215,6 +215,9 @@ def read_config(self, config, **kwargs):
                 "add_threepid_template_text", "add_threepid.txt"
             )
 
+            self.email_password_reset_template_confirmation_html = email_config.get(
+                "password_reset_confirmation_html", "password_reset_confirmation.html"
+            )
```

Comment:
I don't the name of the file needs to be configurable.

What it *does* need is to be documented in the sample config.

Reply:
Looking into https://github.com/matrix-org/synapse/issues/6583 and https://github.com/matrix-org/synapse/issues/7280, I agree that the templates do need a bit of work.

I did this for consistency, but it would introduce yet another config option we'd want to deprecate in the future. I suppose I'll just hardcode the template name here with an explanation then, and at some point we'll deprecate the other options?

I'll update the sample config. I could've sworn I did so already, but it must've got lost on another branch.

Reply:
where these options still exist, they are undocumented and just exist for backwards-compatibility. We don't need to add an option for the new template files.

Reply:
I've removed the new option and documented the template in the sample config.